modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_zephyr.c

```diff
@@ -104,34 +120,139 @@ static bool hw_task_state_set(enum hw_task_state_type expected_state,
 }
 
-static void cc_bind_to_ppi(int32_t cc_num, uint32_t ppi_num)
+static inline uint32_t hw_task_rtc_get_compare_evt_address(int32_t cc_num)
 {
-	uint32_t event_address = z_nrf_rtc_timer_compare_evt_address_get(cc_num);
+#if defined(CONFIG_SOC_NRF5340_CPUNET)
+	return nrf_rtc_event_address_get(HW_TASK_RTC, nrf_rtc_compare_event_get(cc_num));
+#else
+	return z_nrf_rtc_timer_compare_evt_address_get(cc_num);
+#endif
+}
+
+static void hw_task_rtc_cc_bind_to_ppi(int32_t cc_num, uint32_t ppi_num)
+{
+	uint32_t event_address = hw_task_rtc_get_compare_evt_address(cc_num);
 
 	nrfx_gppi_event_endpoint_setup(ppi_num, event_address);
 }
 
-static void cc_unbind(int32_t cc_num, uint32_t ppi_num)
+static void hw_task_rtc_cc_unbind(int32_t cc_num, uint32_t ppi_num)
 {
 	if (ppi_num != NRF_802154_SL_HW_TASK_PPI_INVALID) {
-		uint32_t event_address = z_nrf_rtc_timer_compare_evt_address_get(cc_num);
+		uint32_t event_address = hw_task_rtc_get_compare_evt_address(cc_num);
 
 		nrfx_gppi_event_endpoint_clear(ppi_num, event_address);
 	}
 }
 
-static bool cc_event_check(int32_t cc_num)
+static bool hw_task_rtc_cc_event_check(int32_t cc_num)
 {
-	uint32_t event_address = z_nrf_rtc_timer_compare_evt_address_get(cc_num);
+	uint32_t event_address = hw_task_rtc_get_compare_evt_address(cc_num);
 
 	return *(volatile uint32_t *)event_address;
 }
 
+static void hw_task_rtc_timer_abort(void)
+{
+#if defined(CONFIG_SOC_NRF5340_CPUNET)
+	/* No interrupt disabling/clearing is needed, as HW task does not use interrupts*/
+	nrf_rtc_event_clear(HW_TASK_RTC, NRF_RTC_CHANNEL_EVENT_ADDR(m_hw_task.chan));
+	nrf_rtc_event_disable(HW_TASK_RTC, NRF_RTC_CHANNEL_INT_MASK(m_hw_task.chan));
+#else
+	z_nrf_rtc_timer_abort(m_hw_task.chan);
+#endif
+}
+
+#if defined(CONFIG_SOC_NRF5340_CPUNET)
+static uint32_t zephyr_rtc_ticks_to_hw_task_rtc_counter(uint64_t ticks)
+{
+	uint64_t zephyr_rtc_counter_now = 0;
+	uint32_t hw_task_rtc_counter_now = 0;
+	uint32_t hw_task_rtc_counter_now_2 = 0;
+
+	/**
+	 * The RTC0 counter is read twice - once before and once after reading RTC1.
+	 * If the two reads are different, the procedure is repeated.
+	 * This is to make sure that no RTC tick occured between reading RTC0 and
+	 * RTC1 and so the calculated difference between the timers is correct.
+	 */
+	do
+	{
+		hw_task_rtc_counter_now = nrf_rtc_counter_get(HW_TASK_RTC);
+		zephyr_rtc_counter_now = z_nrf_rtc_timer_read();
+		barrier_dmem_fence_full();
+		barrier_isync_fence_full();
+		hw_task_rtc_counter_now_2 = nrf_rtc_counter_get(HW_TASK_RTC);
+
+	} while(hw_task_rtc_counter_now != hw_task_rtc_counter_now_2);
+
+	/* This function can only be called in close proximity of ticks */
+	assert(ticks >= zephyr_rtc_counter_now ||  zephyr_rtc_counter_now - ticks < RTC_COUNTER_HALF_SPAN);
+	assert(ticks <= zephyr_rtc_counter_now ||  ticks - zephyr_rtc_counter_now < RTC_COUNTER_HALF_SPAN);
+
+	uint32_t diff = (uint32_t) (hw_task_rtc_counter_now - zephyr_rtc_counter_now) & RTC_COUNTER_MAX;
```

Comment:
again you could use `rtc_counter_diff` if you had one

Reply:
Done, although in this case it is not as pretty, because % had to be used to convert `zephyr_rtc_counter_now` to a 32bit value.